lib/spack/spack/relocate.py

```diff
@@ -137,7 +137,7 @@ def get_normalized_elf_rpaths(orig_path_name, rel_rpaths):
     norm_rpaths = list()
     for rpath in rel_rpaths:
         if rpath.startswith('$ORIGIN'):
-            sub = re.sub('$ORIGIN',
+            sub = re.sub(re.escape('$ORIGIN'),
```

Comment:
In general it would be nice to have tests that go along with the bug fix and refer to an issue. See all the tests marked with:
```python
@pytest.mark.regression(number)
```
in the test suite.

Reply:
@eugeneswalker reports errors directly to me in Slack. I will encourage him to file issues for the bugs he finds.

Reply:
@gartung: Issues are fine but we'd like to get the coverage of the buildcache part of the code increased somehow -- see https://codecov.io/gh/spack/spack/list/develop/ and sort the red-dot column descending -- the `buildcache` stuff is probably the least tested (at least w.r.t. unit tests).  Do you have ideas on what we could do?

![Screen Shot 2020-03-19 at 2 51 01 PM](https://user-images.githubusercontent.com/299842/77118281-1941d100-69f1-11ea-8f79-213a1821d523.png)

Reply:
Kind of hard to test most of this without having real buildcaches to test with.

Reply:
Yeah I agree.  I think the build process should create a buildcache for a simple mock package (that actually gets compiled).  If we mock that up, do you think it would help you write tests?

Reply:
yes

Reply:
> Yeah I agree. I think the build process should create a buildcache for a simple mock package (that actually gets compiled). If we mock that up, do you think it would help you write tests?

I'd like to see some test cases like this:
1. Build + cache a couple packages in one prefix
2. Install those packages from the cache into a different prefix
3. Use the packages installed from the cache to build a package

Probably a good idea to cache the packages both with and without the `-r` option to make the rpaths relative.

Reply:
@alalazo  this pr https://github.com/spack/spack/pull/15607 adds a unit test for replace $ORIGIN.